src/valuegraph.jl

```diff
@@ -628,7 +628,57 @@ end
 #  rem_vertex!
 #  ------------------------------------------------------
 
-# TODO
+function LG.rem_vertex!(g::ValGraph, v::Integer)
+    # Procedure: swap v and n, then remove n
+    v in vertices(g) || return false
+    n = nv(g)
+    self_loop_n = false  # true if n is self-looped (see #820)
+
+    # Swap the vertex values of v and n
+    if v != n
+        for i in 1:length(g.vertexvals)
+            g.vertexvals[i][v] = g.vertexvals[i][n]
+        end
+    end
+
+    # Swap the edges and edge values of v and n
+
+    # remove the in_edges incident to v
+    neigs_v = copy(inneighbors(g, v))
+    for s in neigs_v
+        rem_edge!(g, s, v)
```

Comment:
I wonder if it were not better, if we would work directly with the adjacency lists here - after all this method does not have to solve the abstract case, but only has to work for this graph.